### ISSUE TITLE
Fix tests on `feature/trusted-relayer`

### ISSUE THREAD

**Issue by 0xmovses** [movementlabsxyz/movement#914]
We need to get all our integration tests passing before we can proceed with development, not doing so will hinder our development speed as we have seen prior. 

🟢 `client_eth_tests` are all passing 
🟢 `bridge_rest` passing
🟢 `bridge_grpc` passing

- [x] 🟢  `client_mvt_tests test_movement_client_initiate_transfer`
   `Move abort in 0x1::native_bridge_store: ENATIVE_BRIDGE_NOT_ENABLED(0x5)`
- [x] 🟢  `client_mvt_tests test_movement_complete_transfer` `failed execution: FAILED_TO_DESERIALIZE_ARGUMENT`

And naturally, because of the above:
- [x] 🟢  e2e test L1 -> L2  
- [x] 🟢   e2e test L2 -> L1

**Comment by andygolay** [comment]
- I had gotten `test_movemement_client_initiate_transfer` passing last week, see https://github.com/movementlabsxyz/movement/pull/911/commits/0c98d0c9d2f509a50e87dbfe8ac2c689b7894612

- The `FAILED_TO_DESERIALIZE_ARGUMENT` is fixed in this commit I made in PR 911: https://github.com/movementlabsxyz/movement/pull/911/commits/63bbe2d338f0c36c423c5c5b05de3b7f7044ac6d

Maybe at this point it's best to just cherry-pick those fixes from 911 since that branch never got merged last week. I see hundreds of files changed, which is a bit concerning:

<img width="875" alt="image" src="https://github.com/user-attachments/assets/c63a4c3f-92f0-49db-9b06-a7305c9dd053">

I see many various PRs from @musitdev and @l-monninger in #911 , not sure whether all those fixes are needed anymore or if we can just cherry-pick fixes to the Movement client tests?

**Comment by 0xmovses** [comment]
The large diff on #911 Seems to be because `musitdev/update_mvt_test` was cut from main ... because the changes it is trying to merge in are changes from main (namely the name update change). 

If we merge main into the feature this will make the diff more meaningful. I think, it is probably a good time to merge main into feature branch. Then we can get #911 in.

**Comment by 0xmovses** [comment]
With #911 merged into #890 `client_mvt_tests test_movement_client_initiate_transfer` is still failing.

**Comment by andygolay** [comment]
So, the 

> With #911 merged into #903 `client_mvt_tests test_movement_client_initiate_transfer` is still failing.

Looks like the wrong nonce

```
Initiate result: ()
2024-12-03T15:10:33.087192Z  INFO client_mvt_tests: Wait for the Movement Initiated event.
2024-12-03T15:10:33.087699Z  WARN bridge_service::chains::movement::event_monitoring: Check Mvt monitoring loop health channel error: receiving on a closed channel
2024-12-03T15:10:33.091342Z  INFO client_mvt_tests: Received bridge_transfer_id: BridgeTransferId([141, 240, 249, 52, 43, 23, 163, 244, 97, 76, 48, 120, 241, 102, 25, 73, 91, 43, 110, 244, 128, 187, 13, 248, 96, 229, 87, 195, 220, 215, 105, 254])
thread 'test_movement_client_initiate_transfer' panicked at protocol-units/bridge/integration-tests/tests/client_mvt_tests.rs:82:9:
assertion `left == right` failed
  left: Nonce(2)
 right: Nonce(1)
```

Let me see if I can do a small PR to fix it.

**Comment by 0xmovses** [comment]
These tests are all passing as of `09efa951eb3fbf107428fbd4de2fcdcf02d748c4` on `feature/trusted-relayer`
